### ISSUE TITLE
jsonschema update has indirect build dependency on rust compiler

### ISSUE THREAD

**Issue by lengau** [canonical/prometheus-k8s-operator#493]
### Bug Description

The prometheus k8s operator fails to build from a fresh clone

### To Reproduce

1. `git clone git@github.com:canonical/prometheus-k8s-operator.git`
2. `cd prometheus-k8s-operator/`
3. `charmcraft pack --verbose`

### Environment

Using Charmcraft 2.3.0.post12+gitd244e61 (latest/edge),

### Relevant log output

```shell
::    :: Collecting rpds-py>=0.7.1                                                                                                                             
::    ::   Downloading rpds_py-0.8.8.tar.gz (16 kB)                                                                                                            
::    ::   Installing build dependencies: started                                                                                                              
::    ::   Installing build dependencies: finished with status 'error'                                                                                         
::    ::   ERROR: Command errored out with exit status 1:                                                                                                      
::    ::    command: /root/parts/charm/build/staging-venv/bin/python3 /root/parts/charm/build/staging-venv/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-_k8xehix/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- 'maturin>=1.0,<2.0'                                                                                                                                                
::    ::        cwd: None                                                                                                                                      
::    ::   Complete output (65 lines):                                                                                                                         
::    ::   Collecting maturin<2.0,>=1.0                                                                                                                        
::    ::     Downloading maturin-1.1.0.tar.gz (181 kB)                                                                                                         
::    ::     Installing build dependencies: started                                                                                                            
::    ::     Installing build dependencies: finished with status 'done'                                                                                        
::    ::     Getting requirements to build wheel: started                                                                                                      
::    ::     Getting requirements to build wheel: finished with status 'done'                                                                                  
::    ::       Preparing wheel metadata: started                                                                                                               
::    ::       Preparing wheel metadata: finished with status 'done'                                                                                           
::    ::   Collecting tomli>=1.1.0; python_version < "3.11"                                                                                                    
::    ::     Using cached tomli-2.0.1.tar.gz (15 kB)                                                                                                           
::    ::     Installing build dependencies: started                                                                                                            
::    ::     Installing build dependencies: finished with status 'done'                                                                                        
::    ::     Getting requirements to build wheel: started                                                                                                      
::    ::     Getting requirements to build wheel: finished with status 'done'                                                                                  
::    ::       Preparing wheel metadata: started                                                                                                               
::    ::       Preparing wheel metadata: finished with status 'done'                                                                                           
::    ::   Building wheels for collected packages: maturin, tomli                                                                                              
::    ::     Building wheel for maturin (PEP 517): started                                                                                                     
::    ::     Building wheel for maturin (PEP 517): finished with status 'error'                                                                                
::    ::     ERROR: Command errored out with exit status 1:                                                                                                    
::    ::      command: /root/parts/charm/build/staging-venv/bin/python3 /tmp/tmpog3p69_s build_wheel /tmp/tmp7bzppdjd                                          
::    ::          cwd: /tmp/pip-install-kzlounls/maturin                                                                                                       
::    ::     Complete output (33 lines):                                                                                                                       
::    ::     running bdist_wheel                                                                                                                               
::    ::     running build                                                                                                                                     
::    ::     running build_py                                                                                                                                  
::    ::     creating build                                                                                                                                    
::    ::     creating build/lib.linux-x86_64-cpython-38                                                                                                        
::    ::     creating build/lib.linux-x86_64-cpython-38/maturin                                                                                                
::    ::     copying maturin/__main__.py -> build/lib.linux-x86_64-cpython-38/maturin                                                                          
::    ::     copying maturin/__init__.py -> build/lib.linux-x86_64-cpython-38/maturin                                                                          
::    ::     copying maturin/import_hook.py -> build/lib.linux-x86_64-cpython-38/maturin                                                                       
::    ::     running egg_info                                                                                                                                  
::    ::     creating maturin.egg-info                                                                                                                         
::    ::     writing maturin.egg-info/PKG-INFO                                                                                                                 
::    ::     writing dependency_links to maturin.egg-info/dependency_links.txt                                                                                 
::    ::     writing requirements to maturin.egg-info/requires.txt                                                                                             
::    ::     writing top-level names to maturin.egg-info/top_level.txt                                                                                         
::    ::     writing manifest file 'maturin.egg-info/SOURCES.txt'                                                                                              
::    ::     reading manifest file 'maturin.egg-info/SOURCES.txt'                                                                                              
::    ::     reading manifest template 'MANIFEST.in'                                                                                                           
::    ::     warning: no files found matching '*.json' under directory 'src/python_interpreter'                                                                
::    ::     writing manifest file 'maturin.egg-info/SOURCES.txt'                                                                                              
::    ::     running build_ext                                                                                                                                 
::    ::     running build_rust                                                                                                                                
::    ::     error: can't find Rust compiler                                                                                                                   
::    ::                                                                                                                                                       
::    ::     If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.                                                                                               
::    ::                                                                                                                                                       
::    ::     To update pip, run:                                                                                                                               
::    ::                                                                                                                                                       
::    ::         pip install --upgrade pip                                                                                                                     
::    ::                                                                                                                                                       
::    ::     and then retry package installation.                                                                                                              
::    ::                                                                                                                                                       
::    ::     If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.          
::    ::     ----------------------------------------                                                                                                          
::    ::     ERROR: Failed building wheel for maturin                                                                                                          
::    ::     Building wheel for tomli (PEP 517): started                                                                                                       
::    ::     Building wheel for tomli (PEP 517): finished with status 'done'                                                                                   
::    ::     Created wheel for tomli: filename=tomli-2.0.1-py3-none-any.whl size=12758 sha256=b57d53172b3432857125ceaa4b9b2d28e4ae0510c95c5cf031955ada144ba3ff 
::    ::     Stored in directory: /root/.cache/pip/wheels/bb/b6/90/165ed2d4e3933c7567df3ca84de8db1cf47278a1673e8c476c                                          
::    ::   Successfully built tomli                                                                                                                            
::    ::   Failed to build maturin                                                                                                                             
::    ::   ERROR: Could not build wheels for maturin which use PEP 517 and cannot be installed directly                                                        
::    ::   ----------------------------------------                                                                                                            
::    :: ERROR: Command errored out with exit status 1: /root/parts/charm/build/staging-venv/bin/python3 /root/parts/charm/build/staging-venv/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-_k8xehix/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- 'maturin>=1.0,<2.0' Check the logs for full command output.
```

### Additional context

Adding `jsonschema` as a binary package (to prevent downloading and trying to build `rpds-py`) would be a valid workaround.

**Comment by lengau** [comment]
I can confirm the `jsonschema` binary package workaround does work. Checking if just adding `rustc` and `cargo` as build packages works...

**Comment by lucabello** [comment]
Closing the issue as this was solved in #494 :)
